### ISSUE TITLE
2 [B3] Redundant Validations in the Aggregator and Underlying Contracts

### ISSUE THREAD

**Issue by ACassimiro** [soroswap/aggregator#47]
In its current state, transactions requesting swaps to the Aggregator will have their parameters validated at a minimum of two times. When performing token exchanges, there are parameter checks in the Aggregator, the Adapters, and, in the case of the Soroswap router contract, in the router itself (all executed with `check_parameters`). 

While smart contracts should be self-sufficient regarding validations, the Aggregator contract will perform several consecutive contract calls, reflecting potentially expensive transactions. Given the joint operation of these contracts, it makes sense to have these validations removed **once it is guaranteed that they are already performed in the router or liquidity pools, otherwise they should be present at the Adapter level**.

**Comment by esteblock** [comment]
**Recommendation**
While smart contracts should be self-sufficient regarding validations, the Aggregator contract will
perform several consecutive contract calls, reflecting potentially expensive transactions. Given
the joint operation of these contracts, validations regarding the parameters and swap values
should be performed at an Adapter level, if needed, and validations that are confirmed present
at the router or underlying pair contracts can be made absent for the sake of the cost of
transactions.

**Comment by joaquinsoza** [comment]
We have reviewed the audit’s recommendations regarding redundant parameter validations within the Aggregator and Adapter contracts. As a result, we have made the following updates:

- **Removal of Redundant Checks:** Redundant parameter validations have been removed from the Aggregator contract and the Soroswap Adapter contract. These checks were found to be unnecessary since they are already being performed at the protocol level within the respective router or liquidity pool contracts. This change is aimed at optimizing transaction costs by avoiding repetitive validations.
- **Retention of Checks in Phoenix Adapter:** For the Phoenix Adapter, the parameter checks have been retained. Given the possibility of the Phoenix protocol being upgraded in the future, we believe it is prudent to maintain these checks within the Adapter. This ensures an additional layer of security, safeguarding against potential issues that may arise from protocol upgrades.

These adjustments align with the audit’s recommendations while ensuring that security and efficiency are balanced across the different protocols integrated with the Aggregator.
